modAionImpl/test/org/aion/db/AionContractDetailsTest.java

```diff
@@ -336,7 +337,15 @@ public void testExternalStorageTransition() {
         deserialized = deserialize(deserialized.getEncoded(), externalStorage);
 
         for (DataWord key : elements.keySet()) {
-            assertEquals(elements.get(key).toWrapper(), deserialized.get(key.toWrapper()));
+            assertEquals(elements.get(key).toWrapper(), wrapValueFromGet(deserialized.get(key.toWrapper())));
         }
     }
+
+    private static ByteArrayWrapper wrapValueForPut(DataWord value) {
+        return (value.isZero()) ? new ByteArrayWrapper(value.getData()) : new ByteArrayWrapper(value.getNoLeadZeroesData());
+    }
+
+    private static ByteArrayWrapper wrapValueFromGet(ByteArrayWrapper value) {
+        return new ByteArrayWrapper(new DataWord(value.getData()).getData());
```

Comment:
In the final clean-up stage, the `DataWord` should be completely removed from the kernel. Only FVM should know about it.

Reply:
Agreed. It exists here simply because this is a test and we haven't gotten that far into the integration, but soon the kernel will lose visibility on `DataWord` and `DoubleDataWord`